tabular/src/autogluon/tabular/models/xgboost/xgboost_model.py

```diff
@@ -83,6 +85,13 @@ def _fit(self,
         if num_cpus:
             params['n_jobs'] = num_cpus
         max_category_levels = params.pop('proc.max_category_levels', 100)
+        enable_categorical = params.get('enable_categorical', False)
+        if enable_categorical:
+            """Skip one-hot-encoding and pass categoricals directly to XGBoost"""
+            self._ohe = False
+        else:
+            """One-hot-encode categorical features"""
+            self._ohe = True
 
```

Comment:
nit:
```python
self._ohe = not enable_categorical
```

Reply:
Good point, but will keep as is since I have docstrings to describe what True and False mean inline.